src/components/activity/ActivityDirectivesTable.svelte

```diff
@@ -166,17 +166,28 @@
     }
   }
 
-  function canPasteActivityDirectives(): boolean {
-    return plan !== null && hasCreatePermission && canPasteActivityDirectivesFromClipboard(plan);
+  function canPasteActivityDirectives(): Promise<number> {
```

Comment:
Do you mind using async/await for promises? It just makes it a little easier to understand.